crates/starknet-types-core/src/felt/secret_felt.rs

```diff
@@ -2,6 +2,9 @@ use crate::felt::{Felt, FromStrError};
 use lambdaworks_math::errors::ByteConversionError;
 use zeroize::{Zeroize, Zeroizing};
 
+#[cfg(feature = "alloc")]
```

Comment:
This will always be true due to your feature flag configuration:

```
secret_felt = ["alloc", "dep:zeroize"]
```

for this file to be compiled `secret_felt` will be enabled and it now require `alloc`.

So you can just remove this cfg marco altogether

Reply:
The intention was to use the imports only when `no_std` was needed.
I think I got it right this time.